.github/workflows/e2e-tests.yml

```diff
@@ -47,6 +48,11 @@ jobs:
           go-version-file: 'go.mod'
           check-latest: true
 
+      # docker/setup-docker doesn't worth with macos-13 and above runners.
+      - if: matrix.with-containers && runner.os != 'macOS'
```

Comment:
Do we need a container set up here? The e2e that run here do not currently use Kind since the [e2e tests don't run when `cross` is included](https://github.com/sigstore/cosign/blob/d6c40c940fb04cd62156a69b32f46df5a1012bfa/test/e2e_test.go#L16)

Reply:
I think this is the only tests that uses a TSA server.

Reply:
I'm not following why we need the action here, where is the TSA server needed?

Reply:
Needed in test/e2e_tsa_test.go

Reply:
Re-reviewing the change made in test/e2e_tsa_test.go, that's actually changing what it's testing, which is mTLS. Longer term, we can change this test to reuse the TSA spun up by the action, but that will require providing mTLS flags to the server. For now, we'll need to stick with spinning up the TSA via a library.

For e2e_test.go, there's no testing of mTLS so we can use the containerized TSA, but we'll need to refactor a lot of tests to use a common setup function, since each test that uses a TSA spins up a service (like in https://github.com/sigstore/cosign/blob/d6c40c940fb04cd62156a69b32f46df5a1012bfa/test/e2e_test.go#L567-L568). I'm OK with following up later to fix that.

Reply:
Added to https://github.com/sigstore/scaffolding/issues/1555

Reply:
Bumping, the changes in e2e_tsa_test are changing what's being tested to not test mTLS.